server/src/main/java/org/elasticsearch/index/mapper/MetadataFieldMapper.java

```diff
@@ -46,16 +47,24 @@
     }
 
     @SuppressWarnings("rawtypes")
-    public abstract static class Builder<T extends Builder> extends FieldMapper.Builder<T> {
-        public Builder(String name, MappedFieldType fieldType, MappedFieldType defaultFieldType) {
-            super(name, fieldType, defaultFieldType);
+    public abstract static class Builder<T extends Builder<T>> extends FieldMapper.Builder<T> {
+        public Builder(String name, FieldType fieldType) {
+            super(name, fieldType);
+        }
+
+        @Override
+        public T index(boolean index) {
+            if (index == false) {
+                throw new IllegalArgumentException("Metadata fields must be indexed");
```

Comment:
new error?

Reply:
It should be an assertion really - you can't configure metadata fields, so this should never be called.